### PR TITLE
refactor: rename warn event to log, change console logs to log events

### PR DESCRIPTION
**Pull request by brandonocasey**

### CODE REVIEW

File: lib/mp4/caption-parser.js [hunk 1]
```diff
@@ -452,4 +460,6 @@ var CaptionParser = function() {
   this.reset();
 };
 
+CaptionParser.prototype = new Stream();
```
Comment (brandonocasey) [inline lib/mp4/caption-parser.js:463]: Had to make this a stream as we don't have a better way to get listeners right now.

File: test/caption-parser.test.js [hunk 2]
```diff
@@ -10,6 +10,8 @@ var dashInit = segments['dash-608-captions-init.mp4']();
 // This file includes 2 segments data to force a flush
 // of the first caption. The second caption is at 200s
 var dashSegment = segments['dash-608-captions-seg.m4s']();
+var malformedSei = segments['malformed-sei.m4s']();
+var malformedSeiInit = segments['malformed-sei-init.mp4']();
```
Comment (brandonocasey) [inline test/caption-parser.test.js:14]: Added these from the old issue, we should have done it then 🤷🏻

File: lib/mp4/caption-parser.js [hunk 3]
```diff
@@ -97,8 +98,7 @@ var findSeiNals = function(avcStream, samples, trackId) {
         seiNal.pts = lastMatchedSample.pts;
         seiNal.dts = lastMatchedSample.dts;
       } else {
-        // eslint-disable-next-line no-console
-        console.log("We've encountered a nal unit without data. See mux.js#233.");
+        logWarn("We've encountered a nal unit without data. See mux.js#223.");
```
Comment (brandonocasey) [inline lib/mp4/caption-parser.js:101]: The issue number was wrong here. It was actually 223 not 233

**Review by gkatsev — APPROVED** [review]

File: lib/m2ts/metadata-stream.js [hunk 4]
```diff
@@ -91,8 +91,6 @@ var
 MetadataStream = function(options) {
   var
     settings = {
-      debug: !!(options && options.debug),
```
Comment (brandonocasey) [inline lib/m2ts/metadata-stream.js:94]: debug was only used for this console.log, now we just trigger an event at any time and the user can choose what to do.

**Review by gkatsev — APPROVED** [review]

**Pull request merged**